### ISSUE TITLE
Design how users can withdraw a domain request in the action needed state

### ISSUE THREAD

**Issue by amcelya** [cisagov/manage.get.gov#2386]
### Issue description

As an analyst, I will sometimes place a domain request in action needed status to inform a user that their domain request is not yet ready to be fully adjudicated and .gov requires additional action from them to move forward. Sometimes, though, this notification is enough for an organization to determine it will no longer move forward with a domain request at the time. However, the user on the other end is not currently able to delete their domain request when it is in action needed status. Instead, their only option is to edit the request and then resubmit. A user should be able to delete/withdraw their domain at any time prior to approval. 

### Acceptance criteria

- [ ] Adjust the user-facing domain request management to allow users to delete or withdraw their domain while it is in action needed state.
- [ ] Ensure the update is applied to the legacy and enterprise mode layouts.
- [ ] Create dev tickets as necessary, based on Product/Design decisions.

### Additional context

This will reduce the likelihood of abandoned domain requests and will also align the action needed status user-facing options with the other pre-approval request statuses.

**Comment by katypies** [comment]
Putting this in the design parking lot - I suspect we need to initially design any changes to the user flow, and then implement this in the FSM.

**Comment by h-m-f-t** [comment]
>The ticket references the ability to withdraw and delete. But currently, we only allow deletion for "started" tickets. So want to confirm we actually want to add the option to delete for action needed.

Good catch. I support allowing these requests to be withdrawn but not deleted. I'll amend the A/C.

(To be clear, once a request is withdrawn, it can _then_ be deleted by a user. That's fine.)

> I didn't think the help inbox was alerted when that action happens for other requests

It is! I just tested it on stable to confirm.
